### PR TITLE
wallet: Add new wallet constructors

### LINKED ISSUE #3177: Add NewWalletFromBytes function
## Is your feature request related to a problem? Please describe.

There is a function `wallet.NewWalletFromFile`, which is perfect, except for one thing - it uses the file from file system. Often, I already have a wallet like `[]byte` which got somehow. But I still want to unmarshal the wallet from these bytes.

## Describe the solution you'd like
Append `wallet.NewWalletFromBytes(wallet []byte)` function to be able to unmarshal wallet directly from byte slice.

The current function `wallet.NewWalletFromFile` may even use a new one after file reading to unmarshal wallet from bytes.

### PR DESCRIPTION
**Pull request by smallhive**
Closes #3177

### CODE REVIEW

File: pkg/wallet/wallet.go [hunk 1]
```diff
@@ -70,6 +81,16 @@ func NewWalletFromFile(path string) (*Wallet, error) {
 	return wall, nil
 }
 
+// NewWalletFromBytes creates a Wallet from the given byte slice.
+func NewWalletFromBytes(wallet []byte) (*Wallet, error) {
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/wallet/wallet.go:85]: Ref. https://github.com/nspcc-dev/neo-go/issues/3176, although it's a separate problem.

File: pkg/wallet/wallet.go [hunk 2]
```diff
@@ -53,6 +54,16 @@ func NewWallet(location string) (*Wallet, error) {
...
 }
 
+// NewInMemoryWallet creates a new NEO wallet without linking to the read file on file system.
+// If wallet required to be written to the file system, [Wallet.SetPath] should be used to set the path.
+func NewInMemoryWallet() *Wallet {
+	return &Wallet{
+		Version:  walletVersion,
+		Accounts: []*Account{},
+		Scrypt:   keys.NEP2ScryptParams(),
+	}
```
[the remark this answers is not in the document]
Comment (smallhive) [reply pkg/wallet/wallet.go]: Reused

File: pkg/wallet/wallet.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (smallhive) [reply pkg/wallet/wallet.go:92]: Extended documentation

File: pkg/wallet/wallet.go [hunk 3]
```diff
@@ -138,9 +166,16 @@ func (w *Wallet) Path() string {
...
+// SetPath sets the location of the wallet on the filesystem.
+func (w *Wallet) SetPath(path string) {
+	w.path = path
+}
+
 // Save saves the wallet data. It's the internal io.ReadWriter
 // that is responsible for saving the data. This can
 // be a buffer, file, etc..
+//
+// Returns [ErrPathIsEmpty] if wallet pass is not set. See [Wallet.SetPath].
```
Comment (AnnaShaleva) [inline pkg/wallet/wallet.go:178]: s/`pass`/`path`.

**Review by AnnaShaleva — COMMENTED** [review]
Otherwise LGTM.

File: pkg/wallet/wallet.go (hunk 3 above)
Comment (AnnaShaleva) [inline pkg/wallet/wallet.go:178]: OK, while we're here, can we also adjust `Save` documentation itself? To something like:
```
Save saves the wallet data to the file located at the path that was either provided via [NewWalletFromFile] constructor or via [Wallet.SetPath].
```
Not sure whether it's grammatically correct, but we need to specify this behaviour.

Comment (smallhive) [reply pkg/wallet/wallet.go]: Fixed

Comment (smallhive) [reply pkg/wallet/wallet.go]: Updated

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**